### PR TITLE
R2.1 Update fixes #38075

### LINKED ISSUE #38075: Invalid Output shape to Invalid input
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Yes
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: No
- TensorFlow installed from (source or
binary): - TensorFlow 2.1 (source) 
- Python version: - 3.7
version (if compiling from source):
- GCC/Compiler version (if compiling from
source): 
- CUDA/cuDNN version: - GPU model and memory:No

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When`kenerl_size=0` the size of image increases
**Describe the expected behavior**
Value error
**Standalone code to reproduce the issue** 
`import tensorflow as tf`
`inputs=tf.keras.layers.Input(shape=(32, 32, 3))`
`x=tf.keras.layers.Conv2D(64,kernel_size=0)(inputs)`
`x=tf.keras.layers.Flatten()(x)`
`outputs=tf.keras.layers.Dense(64)(x)`
`model=tf.keras.Model(inputs,outputs)`
`model.summary()`

Output:
issue1

### PR DESCRIPTION
**Pull request by maxkaustav**
Update kernel_size zero for value error

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
Can you make it on `master` first and then cherry-pick here please?

We don't update release branches except when doing releases. After a final release the release branch gets updated only for patch releases (which are only done for security purposes)

**Comment by maxkaustav** [comment]
The master brach already has this fixed ,user requested it on tf2.1 so i did it on this branch

**Comment by mihaimaruseac** [comment]
In this case, it will have to wait for until we do a patch release (due to security issues). At that point we will run the CI and investigate if we can safely pull this in.

**Comment by maxkaustav** [comment]
> 
> 
> In this case, it will have to wait for until we do a patch release (due to security issues). At that point we will run the CI and investigate if we can safely pull this in.

Ok and Thank you.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
This is invalid

```
FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/python/keras/layers/convolutional_test.py:170: [E0001(syntax-error), ] expected an indented block
```

File: tensorflow/python/keras/layers/convolutional_test.py [hunk 1]
```diff
@@ -163,6 +163,11 @@ def test_conv2d_constraints(self):
       layer.build((None, 5, 5, 2))
       self.assertEqual(layer.kernel.constraint, k_constraint)
       self.assertEqual(layer.bias.constraint, b_constraint)
+      
+    def test_conv2d_zero_kernel_size(self):
+      kwargs = {'filters': 2, 'kernel_size': 0}
+      with self.assertRaises(ValueError):
+      keras.layers.Conv2D(**kwargs)
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/layers/convolutional_test.py:170]: This line breaks Python syntax

**Pull request closed without merging**